stl/inc/complex

```diff
@@ -76,21 +77,23 @@ public:
         return _Ans;
     }
 
-    static _Ty _Infv() { // return infinity
+    static constexpr _Ty _Infv() { // return infinity
         return numeric_limits<_Ty>::infinity();
     }
 
     static bool _Isinf(_Ty _Left) { // test for infinity
-        double _Tmp = static_cast<double>(_Left);
-        return _CSTD _Dtest(&_Tmp) == _INFCODE;
+        const auto _Tmp  = static_cast<double>(_Left);
+        const auto _Uint = _Bit_cast<uint64_t>(_Tmp);
+        return (_Uint & 0x7fffffffffffffffU) == 0x7ff0000000000000U;
     }
 
-    static bool _Isnan(_Ty _Left) {
-        double _Tmp = static_cast<double>(_Left);
-        return _CSTD _Dtest(&_Tmp) == _NANCODE;
+    static _CONSTEXPR20 bool _Isnan(_Ty _Left) {

```

Comment:
`_CONSTEXPR20` on internal machinery is unusual - does this deserve a comment? (And 206 and 321.)

Reply:
We could actually mark it as `constexpr`, assuming CUDA won't complain that the workaround `_Bit_cast` is never `constexpr` for it. Thoughts?

Reply:
I'd rather not rely on IFNDR when it's easily avoided like this. (I'd rather have `_CONSTEXPR20` with no comment than `constexpr` with no comment.) Admittedly, it's taking longer to devise a reasonable comment than it did to see why this is `_CONSTEXPR20` instead of `constexpr` in the first place, so it wouldn't be the end of the world to have no comment.

Reply:
I think we should leave it as-is then.